hwy/ops/rvv-inl.h

```diff
@@ -3094,6 +3098,150 @@ HWY_API V Broadcast(const V v) {
   return TableLookupLanes(v, idx);
 }
 
+// ------------------------------ BroadcastLane
+#ifdef HWY_NATIVE_BROADCASTLANE
+#undef HWY_NATIVE_BROADCASTLANE
+#else
+#define HWY_NATIVE_BROADCASTLANE
+#endif
+
+template <int kLane, class V,
+          hwy::EnableIf<(sizeof(TFromV<V>) != 1 || kLane <= 255)>* = nullptr>
+HWY_API V BroadcastLane(V v) {
+  static_assert(0 <= kLane && kLane < HWY_MAX_LANES_V(V), "Invalid lane");
+  const DFromV<V> d;
+  const RebindToUnsigned<decltype(d)> du;
+  using TU = TFromD<decltype(du)>;
+  return TableLookupLanes(v, Set(du, static_cast<TU>(kLane)));
```

Comment:
RVV's rgather_vx is likely much more efficient here. Let's use that in all three overloads here?
https://github.com/riscv-non-isa/rvv-intrinsic-doc/blob/master/auto-generated/policy_funcs/intrinsic_funcs/10_vector_permutation_functions.md

Reply:
Re-implemented BroadcastLane using the RVV rgather_vx intrinsics.